File: examples/sharing/share-remove.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Remove access by user ID, or cancel a member/pending grant by email.
export async function removeShare(
  client: JsBaoClient,
  documentId: string,
  userId: string,
) {
  // #region example
  await client.documents.removePermission(documentId, userId);
  await client.documents.removePermission(documentId, { email: "alice@example.com" });
  // #endregion example
}

File: examples/sharing/share-remove.swift
import JsBaoClient

// Remove access by user ID, or cancel a member/pending grant by email.
func removeShare(
  client: JsBaoClient,
  documentId: String,
  userId: String
) async throws {
  // #region example
  try await client.documents.removePermission(documentId: documentId, .userId(userId))
  try await client.documents.removePermission(documentId: documentId, .email("alice@example.com"))
  // #endregion example
}
